vms/docker/start-server.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+if [ ! -f /standardebooks.org/web/config/ssl/standardebooks.test.crt ]; then
+	openssl req -x509 -nodes -days 99999 -newkey rsa:4096 -subj "/CN=standardebooks.test" -keyout /standardebooks.org/web/config/ssl/standardebooks.test.key -sha256 -out /standardebooks.org/web/config/ssl/standardebooks.test.crt
+fi
+
+cd /standardebooks.org/web
+composer install
+
+ln -s /standardebooks.org/web/config/apache/standardebooks.test.conf /etc/apache2/sites-available/
+ln -s /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/*/cli/conf.d/
+ln -s /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/*/fpm/conf.d/
+ln -s /standardebooks.org/web/config/php/fpm/standardebooks.test.conf /etc/php/*/fpm/pool.d/
+a2ensite standardebooks.test
+/etc/init.d/apache2 start
+service php7.4-fpm restart
```

Comment:
So this is a super-stripped back container. The Docker idea is that each container effectively does one thing with the minimal resources it needs, and if you need more services (i.e. if we had a database for the site) then you’d spin that up as a separate container and expose the necessary interfaces instead. Indeed, there’s Docker Compose for orchestrating multiple containers for your single app, though I’ve avoided that for this.

Because you’re not running multiple systems on the single container, they seem to have gone with a more stripped back approach:

> System has not been booted with systemd as init system (PID 1). Can't operate.

Reply:
Started Apache in the same way as PHP in https://github.com/standardebooks/web/pull/108/commits/4718db4e8f2feeff806873730892e7668b25537a.